From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Tue, 31 May 2016 19:05:56 +0300
Subject: [PATCH 310/453] IB/core: fix an error code in ib_core_init()
Patch-mainline: v4.7-rc5
Git-commit: da1f857be62ca0472024da37eab068b3b8ce0a15
References: FATE#321231 FATE#321473 FATE#322153 FATE#322149

We should return the error code if ib_add_ibnl_clients() fails.  The
current code returns success.

Fixes: 735c631ae99d ('IB/core: Register SA ibnl client during ib_core initialization')
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Reviewed-by: Mark Bloch <markb@mellanox.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/core/device.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/infiniband/core/device.c b/drivers/infiniband/core/device.c
index 9c869ef09474..8febc09b8feb 100644
--- a/drivers/infiniband/core/device.c
+++ b/drivers/infiniband/core/device.c
@@ -1033,7 +1033,8 @@ static int __init ib_core_init(void)
 		goto err_mad;
 	}
 
-	if (ib_add_ibnl_clients()) {
+	ret = ib_add_ibnl_clients();
+	if (ret) {
 		pr_warn("Couldn't register ibnl clients\n");
 		goto err_sa;
 	}
-- 
1.8.5.6

